### PR TITLE
Cosmos OLTP spark: added support for `spark.cosmos.preferredRegionsList` config

### PR DESCRIPTION
**Pull request by moderakh**
- added new config `spark.cosmos.preferredRegionsList` sample value `[eastus, west us1]` or `eastus, westus1`
- improved the logic for config enum parsing. added a unified `CosmosConfigEntry.parseEnumeration(.)` and refactored existing code to use that.
- `"spark.cosmos.partitioning.STRATEGY" -> ""` empty config for default is not a valid config anymore. The value needs to be explicitly passed.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md [hunk 1]
```diff
@@ -20,11 +20,10 @@ Configuration Reference:
 | `spark.cosmos.useGatewayMode`      | `false`    | Use gateway mode for the client operations  |
 | `spark.cosmos.read.forceEventualConsistency`  | `true`    | Makes the client use Eventual consistency for read operations instead of using the default account level consistency |
 | `spark.cosmos.applicationName`      | None    | Application name  |
-
+| `spark.cosmos.preferredRegionsList`      | None    | Preferred regions list to be used for a multi region Cosmos DB account. This is a comma separated value (e.g., `[eastus,westus]`) provided preferred regions will be used as hint. You should use a collocated spark cluster with your Cosmos DB account and pass the spark cluster region as preferred region.  |
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:23]: Would it make sense to add the link to the azure region identifiers: https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.locationnames?view=azure-dotnet

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:23]: adding that now. Thanks @FabianMeiswinkel

Comment (ealsur) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:23]: One interesting difference. The LocationNames use the DisplayName property of the ARM Location, like West US (with spaces), while this spark doc is showing the Name property (lowercase, no spaces). Is this expected? Do both work or just the Name for Java SDK?

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientConfiguration.scala [hunk 2]
```diff
@@ -31,7 +32,8 @@ private[spark] object CosmosClientConfiguration {
       cosmosAccountConfig.key,
       applicationName,
       cosmosAccountConfig.useGatewayMode,
-      useEventualConsistency)
+      useEventualConsistency,
+      cosmosAccountConfig.preferredRegionsList.getOrElse(Array[String]()))
```
Comment (ealsur) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientConfiguration.scala:36]: Can this default be at the AccountConfig level when it's parsed?

**Review by ealsur — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:23]: all work, we do proper name normalization. trimming, lower-casing ...

Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md]: done.

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientConfiguration.scala (hunk 2 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientConfiguration.scala]: valid point. thanks Matias.

I changed to code to not have default outside of CosmosConfig.

**Pull request merged**